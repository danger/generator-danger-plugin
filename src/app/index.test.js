import assert from 'yeoman-assert'
import helpers from 'yeoman-test'
import fs from 'fs'

jest.mock('npm-name', () => () => Promise.resolve(true))
jest.mock('./values', () => ({
  defaultName: () => Promise.resolve('Git Name'),
  defaultEmail: () => Promise.resolve('git@git.com'),
}))

function readFile(filename, json) {
  const file = fs.readFileSync(filename, 'utf8')
  return json ? JSON.parse(file) : file
}

describe('generator:app', () => {
  it('creates expected files', async () => {
    await helpers.run(__dirname).withPrompts({
      pluginName: 'danger-plugin-fun-time',
      description: 'Danger plugin that tells you to have a fun time',
      authorName: 'Macklin Underdown',
      authorEmail: 'email@example.com',
    })
    assert.file([
      'package.json',
      'CODE_OF_CONDUCT.md',
      'LICENSE.md',
      '.editorconfig',
      '.gitignore',
      '.npmignore',
    ])
  })
  describe('CODE_OF_CONDUCT.md', () => {
    it('contains author email', async () => {
      await helpers.run(__dirname).withPrompts({
        pluginName: 'danger-plugin-fun-time',
        description: 'Danger plugin that tells you to have a fun time',
        authorName: 'Macklin Underdown',
        authorEmail: 'email@example.com',
      })
      assert.fileContent('CODE_OF_CONDUCT.md', 'email@example.com')
    })
  })
  describe('LICENSE.md', () => {
    it('contains the current year and author name', async () => {
      Date.prototype.getUTCFullYear = jest.fn(() => 2017)
      await helpers.run(__dirname).withPrompts({
        pluginName: 'danger-plugin-fun-time',
        description: 'Danger plugin that tells you to have a fun time',
        authorName: 'Macklin Underdown',
        authorEmail: 'email@example.com',
      })
      assert.fileContent('LICENSE.md', /2017 Macklin Underdown/g)
    })
  })
  describe('package.json', () => {
    it('fills package.json with correct information', async () => {
      await helpers.run(__dirname).withPrompts({
        pluginName: 'danger-plugin-fun-time',
        description: 'Danger plugin that tells you to have a fun time',
        authorName: 'Macklin Underdown',
        authorEmail: 'email@example.com',
      })
      const pkg = readFile('package.json', true)
      expect(pkg).toMatchSnapshot()
    })
    it('uses git user.name as default name prompt value', async () => {
      await helpers.run(__dirname).withPrompts({
        pluginName: 'danger-plugin-fun-time',
        description: 'Danger plugin that tells you to have a fun time',
      })
      const pkg = readFile('package.json', true)
      expect(pkg.author.name).toEqual('Git Name')
    })
    it('uses git user.email as default email prompt value', async () => {
      await helpers.run(__dirname).withPrompts({
        pluginName: 'danger-plugin-fun-time',
        description: 'Danger plugin that tells you to have a fun time',
      })
      const pkg = readFile('package.json', true)
      expect(pkg.author.email).toEqual('git@git.com')
    })
  })
})
