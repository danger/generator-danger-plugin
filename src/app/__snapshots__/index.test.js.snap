// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator:app CODE_OF_CONDUCT.md contains author email 1`] = `
"# Contributor Covenant Code of Conduct

## Our Pledge

In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.

## Our Standards

Examples of behavior that contributes to creating a positive environment include:

* Using welcoming and inclusive language
* Being respectful of differing viewpoints and experiences
* Gracefully accepting constructive criticism
* Focusing on what is best for the community
* Showing empathy towards other community members

Examples of unacceptable behavior by participants include:

* The use of sexualized language or imagery and unwelcome sexual attention or advances
* Trolling, insulting/derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or electronic address, without explicit permission
* Other conduct which could reasonably be considered inappropriate in a professional setting

## Our Responsibilities

Project maintainers are responsible for clarifying the standards of acceptable behavior and are expected to take appropriate and fair corrective action in response to any instances of unacceptable behavior.

Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct, or to ban temporarily or permanently any contributor for other behaviors that they deem inappropriate, threatening, offensive, or harmful.

## Scope

This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team at email@example.com. The project team will review and investigate all complaints, and will respond in a way that it deems appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.

Project maintainers who do not follow or enforce the Code of Conduct in good faith may face temporary or permanent repercussions as determined by other members of the project's leadership.

## Attribution

This Code of Conduct is adapted from the [Contributor Covenant][homepage], version 1.4, available at [http://contributor-covenant.org/version/1/4][version]

[homepage]: http://contributor-covenant.org
[version]: http://contributor-covenant.org/version/1/4/
"
`;

exports[`generator:app LICENSE.md contains the current year and author name 1`] = `
"Copyright 2017 Macklin Underdown

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the \\"Software\\"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED \\"AS IS\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
"
`;

exports[`generator:app package.json fills package.json with correct JS information 1`] = `
Object {
  "author": Object {
    "email": "email@example.com",
    "name": "Macklin Underdown",
  },
  "bugs": Object {
    "url": "https://github.com/macklinu/danger-plugin-fun-time/issues",
  },
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "description": "Danger plugin that tells you to have a fun time",
  "devDependencies": Object {
    "@babel/cli": "^7.0.0",
    "@babel/core": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "babel-jest": "^25.1.0",
    "commitizen": "^4.0.3",
    "cz-conventional-changelog": "^3.1.0",
    "esdoc": "^1.1.0",
    "esdoc-standard-plugin": "^1.0.0",
    "husky": "^4.2.3",
    "jest": "^25.1.0",
    "lint-staged": "^10.0.7",
    "prettier": "^1.19.1",
    "semantic-release": "^17.0.3",
    "typescript": "^3.6.4",
    "typings-tester": "^0.3.2",
    "validate-commit-msg": "^2.14.0",
  },
  "engines": Object {
    "node": ">=10.0.0",
  },
  "homepage": "https://github.com/macklinu/danger-plugin-fun-time#readme",
  "husky": Object {
    "hooks": Object {
      "commit-msg": "validate-commit-msg",
      "pre-commit": "lint-staged",
    },
  },
  "keywords": Array [
    "danger",
    "danger-plugin",
    "fun",
    "time",
  ],
  "license": "MIT",
  "lint-staged": Object {
    "*.js": Array [
      "prettier --single-quote --trailing-comma=all --no-semi --write",
      "git add",
    ],
  },
  "main": "dist/index.js",
  "name": "danger-plugin-fun-time",
  "repository": Object {
    "type": "git",
    "url": "https://github.com/macklinu/danger-plugin-fun-time.git",
  },
  "scripts": Object {
    "build": "babel src --out-dir dist --ignore *.test.js",
    "commit": "git-cz",
    "docs": "esdoc -c .esdoc.json",
    "lint": "tslint \\"src/**/*.ts\\"",
    "predocs": "rm -rf docs/",
    "prepare": "npm run build",
    "prettier": "prettier",
    "prettier-project": "npm run prettier-write -- 'src/**/*.{ts,tsx}'",
    "prettier-write": "npm run prettier -- --parser typescript --no-semi --trailing-comma es5 --write --print-width 120",
    "semantic-release": "semantic-release",
    "test": "jest",
  },
  "version": "0.0.0-development",
}
`;

exports[`generator:app package.json fills package.json with correct TS information 1`] = `
Object {
  "author": Object {
    "email": "email@example.com",
    "name": "Macklin Underdown",
  },
  "bugs": Object {
    "url": "https://github.com/macklinu/danger-plugin-fun-time/issues",
  },
  "config": Object {
    "commitizen": Object {
      "path": "cz-conventional-changelog",
    },
  },
  "description": "Danger plugin that tells you to have a fun time",
  "devDependencies": Object {
    "@types/jest": "^25.1.2",
    "@types/node": "^12.7.12",
    "commitizen": "^4.0.3",
    "cz-conventional-changelog": "^3.1.0",
    "danger": "*",
    "esdoc": "^1.1.0",
    "esdoc-standard-plugin": "^1.0.0",
    "husky": "^4.2.3",
    "jest": "^25.1.0",
    "lint-staged": "^10.0.7",
    "prettier": "^1.19.1",
    "semantic-release": "^17.0.3",
    "ts-jest": "^25.1.0",
    "tslint": "^6.0.0",
    "typescript": "^3.6.4",
    "validate-commit-msg": "^2.14.0",
  },
  "engines": Object {
    "node": ">=10.0.0",
  },
  "homepage": "https://github.com/macklinu/danger-plugin-fun-time#readme",
  "husky": Object {
    "hooks": Object {
      "commit-msg": "validate-commit-msg",
      "pre-commit": "lint-staged",
    },
  },
  "jest": Object {
    "moduleFileExtensions": Array [
      "ts",
      "tsx",
      "js",
    ],
    "testPathIgnorePatterns": Array [
      "\\\\.snap$",
      "<rootDir>/node_modules/",
    ],
    "testRegex": "(.test)\\\\.(ts|tsx)$",
    "transform": Object {
      ".(ts|tsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js",
    },
  },
  "keywords": Array [
    "danger",
    "danger-plugin",
    "fun",
    "time",
  ],
  "license": "MIT",
  "lint-staged": Object {
    "*.@(ts|tsx)": Array [
      "tslint --fix",
      "npm run prettier-write --",
    ],
  },
  "main": "dist/index.js",
  "name": "danger-plugin-fun-time",
  "repository": Object {
    "type": "git",
    "url": "https://github.com/macklinu/danger-plugin-fun-time.git",
  },
  "scripts": Object {
    "build": "tsc",
    "commit": "git-cz",
    "docs": "esdoc -c .esdoc.json",
    "lint": "tslint \\"src/**/*.ts\\"",
    "predocs": "rm -rf docs/",
    "prepare": "npm run build",
    "prettier": "prettier",
    "prettier-project": "npm run prettier-write -- 'src/**/*.{ts,tsx}'",
    "prettier-write": "npm run prettier -- --parser typescript --no-semi --trailing-comma es5 --write --print-width 120",
    "semantic-release": "semantic-release",
    "test": "jest",
  },
  "types": "dist/index.d.ts",
  "version": "0.0.0-development",
}
`;

exports[`generator:app src/ generates source file based on plugin name 1`] = `
"/**
 * Danger plugin that tells you to have a fun time
 */
export default function funTime() {
  // TODO
}
"
`;

exports[`generator:app src/ generates test file based on plugin name 1`] = `
"describe('funTime()', () => {
  it('does something')
})
"
`;

exports[`generator:app types/ generates TypeScript test file 1`] = `
"import funTime from './'

funTime()
"
`;

exports[`generator:app types/ generates TypeScript type defintion 1`] = `
"export default function funTime(): void
"
`;
